Rework contribution docs edits

blackbaud-johnly opened the pull request:
Updates to https://github.com/blackbaud/skyux2-docs/pull/622 to address https://github.com/blackbaud/skyux2-docs/issues/617.

Blackbaud-ErikaMcVey commented on src/app/contribute/overview/index.html:
```diff
@@ -23,80 +17,85 @@
+          After you <a stacheRouterLink="/contribute/overview/file-issue">file an issue</a>, discuss the architecture and implementation details, requirements, and timelines with the SKY UX team. We will assign a developer to your issue to help answer questions.
+        </p>
+        <p>
+          Keep in mind that the Blackbaud design council must approve changes that impact the UI, such as new components or changes to existing components. Contributions that do not have design approval must include detailed design specifications. The design approval process can take significant time if components are complex or designs require significant changes, so account for that in your dev ...
```
I like this blurb but I think it might make more sense in the previous step, filing an issue! The design should be part referenced in the issue and I think is often necessary for the implementation/architecture discussion

blackbaud-johnly replied:
I'll move this up and also tighten step 1 up so that it isn't too long.

Blackbaud-ErikaMcVey commented on src/app/contribute/contribute-action-button.component.ts:
```diff
@@ -7,16 +7,16 @@ import { Component } from '@angular/core';
   public routes = [
     {
-      name: 'Overview',
+      name: 'Contribution process',
```
the route for this is still `https://developer.blackbaud.com/skyux/contribute/overview`, do we want to change that to `https://developer.blackbaud.com/skyux/contribute/process` or something?

blackbaud-johnly replied:
I don't have a strong preference, but it did occur to me that maybe we should change this. Since you seem to have the same thought, I'll go ahead and update the path and all the related links.

Blackbaud-DeniseCisneros commented on src/app/contribute/overview/guidelines/index.html:
```diff
@@ -0,0 +1,102 @@
+    API changes
+  </stache-page-anchor>
+  <p>
+    For API changes that affect inputs and outputs for a component, we recommend that you provide your potential (not working) documentation and demo to the SKY UX team for review. This lets us head off API issues before implementation. Contributors should keep in mind that the API can change in the course of development.
```
Can we switch out "head off" to something like "review" or "address"?

blackbaud-johnly replied:
Good call. I'll change it to "address"

Blackbaud-DeniseCisneros commented on src/app/contribute/overview/guidelines/index.html:
```diff
@@ -0,0 +1,102 @@
+    Pull requests
+  </stache-page-anchor>
+  <p>
+    For large changes that include multiple components, submit changes in several small pull requests instead of one large pull request. We definte large as more than 50 files.
```
Small typo. Define

blackbaud-johnly replied:
Good catch. Thanks!

blackbaud-johnly commented on src/app/contribute/contribute-action-button.component.ts:
```diff
@@ -7,16 +7,16 @@ import { Component } from '@angular/core';
       path: '/contribute/overview',
       icon: 'file',
-      summary: 'An overview of contributing to SKY UX.'
+      summary: 'Learn how to contribute to SKY UX, and review our contribution guidelines.'
```
I think you could go either way here. Technically, it is a compound sentence joining commands. The commands both have the same implicit subject ("You"), so you could definitely argue for getting rid of the comma. My reasoning for including the comma is that the comma could make it seem like the implicit subject and the first predicate both apply to the whole sentence. (i.e., You learn how to contribute to SKY UX and learn how to review our contribution guidelines). But maybe I'm just overthinking it. I can definitely delete the comma if you think it flows better without it. Just let me know what you think.

Blackbaud-BobbyEarl replied:
Just want to add how much I appreciate this dialogue and the technical grammar explanation.

blackbaud-johnly replied:
I appreciate your appreciation. ... And presumably with the more sincerity than your appreciation of the dialogue. ;-)

blackbaud-johnly commented on src/app/contribute/overview/get-code/index.html:
```diff
@@ -0,0 +1,27 @@
+      Create a branch off of <sky-code>master</sky-code>. Name it after the feature to contribute, and use kebab-case (<sky-code>my-new-feature</sky-code>).
+    </li>
+    <li>
+      Clone your repo locally and run <sky-code>nvm use</sky-code> to ensure you have the preferred version of Node.js installed. Then run <sky-code>npm install</sky-code> to install all required dependencies. For more information on prerequisites, see <a stacheRouterLink="/learn/get-started/prereqs">SKY UX prerequisites</a>.
```
@Blackbaud-BobbyEarl, what do you think? Should we switch `npm install` to `skyux install` here? I assume so since I think these docs predate `skyux install`, but I just want to make sure.

Blackbaud-BobbyEarl replied:
In this particular case, `npm install` is still the correct instruction.

blackbaud-johnly replied:
OK. Thanks! I'll leave it as is.

blackbaud-johnly commented:
@Blackbaud-ErikaMcVey, I think I addressed all three comments you made. @Blackbaud-DeniseCisneros, I addressed 2 items from you, confirmed with Bobby to leave one as is, and replied to the fourth; let me know if you think we should remove that comma, and I'll get right on that. ... And let me know if either of you has any additional suggestions. Thanks!

Blackbaud-DeniseCisneros commented:
I don't have a very strong opinion about the comma, and I'm cool with leaving it. Let's merge this thing, woohoo! 🎉
